secure/storage/src/kv_storage.rs

```diff
@@ -22,8 +24,8 @@ pub trait KVStorage: Send + Sync {
 
     /// Resets and clears all data held in the storage engine.
     /// Note: this should only be exposed and used for testing. Resetting the storage engine is not
-    /// something that should be supported in production.
-    #[cfg(any(test, feature = "testing"))]
+    /// something that should be supported in production. In production, this
+    /// operation should be a no-op.
     fn reset_and_clear(&mut self) -> Result<(), Error>;
```

Comment:
this is not good, I would prefer that if this is implemented in production that it panics.... does enum_dispatch not work effectively for feature flags?

why can we not retain the existing behavior?

Reply:
Well, quite simply, `enum_dispatch` does not support that yet.

Reply:
I have a PR open to fix it, which should help lighten the present PR: https://gitlab.com/antonok/enum_dispatch/-/merge_requests/15

Reply:
No longer a problem with enum_dispatch 0.3.1.